ps_linklist.php

```diff
@@ -58,57 +74,87 @@ public function __construct()
         $this->description = $this->trans('Adds a block with several links.', array(), 'Modules.Linklist.Admin');
         $this->secure_key = Tools::encrypt($this->name);
 
-        $this->ps_versions_compliancy = array('min' => '1.7.1.0', 'max' => _PS_VERSION_);
+        $this->ps_versions_compliancy = array('min' => '1.7.5.0', 'max' => _PS_VERSION_);
         $this->templateFile = 'module:ps_linklist/views/templates/hook/linkblock.tpl';
 
         $this->linkBlockPresenter = new LinkBlockPresenter(new Link(), $this->context->language);
-        $this->linkBlockRepository = new LinkBlockRepository(Db::getInstance(), $this->context->shop, $this->context->getTranslator());
+        $this->legacyBlockRepository = new LegacyLinkBlockRepository(Db::getInstance(), $this->context->shop, $this->context->getTranslator());
     }
 
     public function install()
     {
-        return parent::install()
-            && $this->installTab()
-            && $this->linkBlockRepository->createTables()
-            && $this->linkBlockRepository->installFixtures()
+        if (!parent::install()) {
+            return false;
+        }
+        $installed = true;
+
+        $errors = $this->getRepository()->createTables();
+        if (!empty($errors)) {
+            $this->addModuleErrors($errors);
+            $installed = false;
+        }
+
+        $errors = $this->getRepository()->installFixtures();
+        if (!empty($errors)) {
+            $this->addModuleErrors($errors);
+            $installed = false;
+        }
+
+        if ($installed
             && $this->registerHook('displayFooter')
-            && $this->registerHook('actionUpdateLangAfter');
+            && $this->registerHook('actionUpdateLangAfter')
+            && $this->installTab()) {
+            return true;
+        }
+
+        parent::uninstall();
+
+        return false;
     }
 
     public function uninstall()
     {
-        return parent::uninstall()
-            && $this->uninstallTab()
-            && $this->linkBlockRepository->dropTables();
+        $uninstalled = true;
+        if (!$this->uninstallTab()) {
+            $this->_errors[] = $this->trans('Could not remove Tab.', array(), 'Admin.Modules.Notification');
+            $uninstalled = false;
+        }
+        $errors = $this->getRepository()->dropTables();
+        if (!empty($errors)) {
+            $this->addModuleErrors($errors);
+            $uninstalled = false;
+        }
+
+        return $uninstalled && parent::uninstall();
     }
 
     public function installTab()
```

Comment:
Why do not use the new method for create Tabs ?
http://doc.prestashop.com/display/PS17/Adding+module+links+in+the+back-office+side+menu

Introduced in https://github.com/PrestaShop/PrestaShop/pull/5922

Reply:
I didn't know this feature, much better!

Reply:
Watch out with this part. Not sure it's working for modules that are installed during installation process.

Reply:
We hope to see some improvements with 1.7.5, thanks to https://github.com/PrestaShop/PrestaShop/pull/10296

Reply:
@Matt75 the `$tabs` does not seem to work btw

Reply:
@jolelievre We can have a check together today if you want. We need the @Quetzacoalt91 patch to make `$tabs` working during PrestaShop installation :)

Reply:
yes I tested it via the module page by uninstalling it and reinstalling it, I didn't try during the installation process because I don't really know how to do this :/
about the $tabs feature it actually didn't work and I could not find any modules using this feature

Reply:
ERROR                                                 17:55:15 | app | Class "AdminLinkWidgetController" not found in controllers/admin
-- | -- | --


Obviously ... The system only works with the legacy controllers

Reply:
so I don't have a choice but to create the tab "manually" ?

Reply:
For now, yes.

Tab registration need to be updated to work with symfony routes.